### PR TITLE
[stable/external-dns] Added AWS access key environment variables

### PR DESCRIPTION
**Pull request by micahhausler**
* Applicable resources are now namespaced

### CODE REVIEW

**Comment by craigtracey** [comment]
Needed in my environment as well.

:+1: LGTM

File: stable/external-dns/templates/deployment.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metadata:
   labels: {{ include "external-dns.labels" . | indent 4 }}
     chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
   name: {{ template "external-dns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (mgoodness) [inline stable/external-dns/templates/deployment.yaml]: We don't set the namespace, delegating that to Tiller.

File: stable/external-dns/templates/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{{- if and .Values.aws.secret_key .Values.aws.access_key -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels: {{ include "external-dns.labels" . | indent 4 }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: {{ template "external-dns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (mgoodness) [inline stable/external-dns/templates/secret.yaml]: No namespace

File: stable/external-dns/templates/service.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ metadata:
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
   name: {{ template "external-dns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (mgoodness) [inline stable/external-dns/templates/service.yaml]: No namespace

File: stable/external-dns/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -5,4 +5,5 @@ metadata:
   labels: {{ include "external-dns.labels" . | indent 4 }}
     chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
   name: {{ template "external-dns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (mgoodness) [inline stable/external-dns/templates/serviceaccount.yaml]: No namespace

**Review by mgoodness — CHANGES_REQUESTED** [review]
Drop the Namespace fields & rebase. Should be ready after that.

**Comment by micahhausler** [comment]
@mgoodness fixed!

**Review by mgoodness — APPROVED** [review]

**Pull request merged**